### PR TITLE
Issue #3318349 by agami4: Improve class for the `border-radius` style from theme settings

### PR DESCRIPTION
**Pull request by Agamich-FE**
## Problem
When we have a button group, we do not have a `border-radius` style from theme settings.

## Solution
Change selector from `not(.btn-group) > .btn` to `.btn`.

## Issue tracker

- https://www.drupal.org/project/social/issues/3318349
- https://getopensocial.atlassian.net/browse/PROD-22926

## Theme issue tracker
- https://www.drupal.org/project/socialblue/issues/3318340

## How to test
- [ ] Go to the group and press the `join` button.
- [ ] After reloading the page check the `border-radius` button.

## Release notes
All buttons now have correct `border-radius` style from the theme settings.

## Screenshot
image

## Change Record
N/A

## Translations
N/A

### CODE REVIEW

**Review by ribel — APPROVED** [review]
Changes looks good to me.

**Review by tbsiqueira — APPROVED** [review]

**Review by Kingdutch — CHANGES_REQUESTED** [review]

File: modules/custom/improved_theme_settings/improved_theme_settings.module
Comment (agami4) [reply modules/custom/improved_theme_settings/improved_theme_settings.module:182]: Hey @Kingdutch, sorry for the long reply. I added this improvement because `btn-group` looks like a one-button with a dropdown arrow inside this button.
Screenshot 2022-11-15 at 15 26 51
I checked all pages on the ECI demo env and did not find other cases.

Comment (Kingdutch) [reply modules/custom/improved_theme_settings/improved_theme_settings.module:182]: @agami4 looking at the Bootstrap documentation I can come up with some scenarios where our new code may cause bugs: https://getbootstrap.com/2.3.2/components.html#buttonGroups

I'm assuming the original `.btn-group` exclusion was added because you may have [button1][button2] without any spatial separation in a button group. If you apply any border radius to them then that would become (button1)(button2) which may not look right.

It looks like specifically here the bug is not about `.btn` in a `.btn-group` but about a `.dropdown-toggle` in that button group. So perhaps we could make the code more specific so we only target our new use-case: `:not(.btn-group) > .btn, .btn.dropdown-toggle`. That way we don't break the example case for which we made an exception (even if it's not currently findable on ECI demo) but we do fix the broken case.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3193-p2q5uufdom1zlcoleauxenuk10ec2pyn.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/635fdb0b8cecf091c95cba7e

File: modules/custom/improved_theme_settings/improved_theme_settings.module
Comment (agami4) [reply modules/custom/improved_theme_settings/improved_theme_settings.module:182]: @Kingdutch, I revert the changes but added a specific class for buttons on the buttons group just for the case where I have a bug. Please check.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3193-p2q5uufdom1zlcoleauxenuk10ec2pyn.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/635fdb0b8cecf091c95cba7e

**Review by nkoporec — APPROVED** [review]
The latest changes looks good to me, but will wait for final approval from @Kingdutch before merging this one.

**Review by Kingdutch — APPROVED** [review]
We'll probably have to fix the exception at some point in the future (since they're usually a sign of inconsistency in our design system). However, I think in this case it's the safest option to fix the issue at hand without risk of causing other problems elsewhere, so happy to take that route for now :)

**Comment by nkoporec** [comment]
🍒 picked to 11.5.x, 11.6.x, 11.7.x

**Pull request merged**
